### ISSUE TITLE
Device totally unresponsive

### ISSUE THREAD

**Issue by YosypC** [linux-surface/linux-surface#1173]
### Environment

- Hardware model: Pro 8
- Kernel version: Linux yosyp-surface-fedora 6.3.7-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jun  9 15:21:11 UTC 2023 x86_64 GNU/Linux
- Distribution: Fedora 38

### Debug

[debug.zip](https://github.com/linux-surface/linux-surface/files/11815340/debug.zip) containing dmesg with and without -k, journalctl since yesterday.

### Description

Device woken up from sleep after a night. Done some browsing, opened the calculator, didn't work: the touchscreen was totally unresponsive on the gnome calc ALONE, worked on Firefox but not well on the ARC bar on top of my screen (I believed it was a bug of the GNOME's extension, but it may not be!). I ended up rebooting the device normally from GNOME, and successfully logged in with the touchscreen. After reopening Firefox and the calculator, I shot this [video](https://github.com/linux-surface/linux-surface/files/11815381/20230621_101101.zip).
You can see me pressing the button several times.

Shortly after I tried attaching the keyboard, the cursor was alive but I wasn't able to do anything. Hard shut off with the button was needed.

Note that after the boot, GRUB appeared (it never did) and I selected the latest kernel. I am now typing on my keyboard, and the touch doesn't work at all (might be related to #1171).

**Comment by StollD** [comment]
`6.3.7-200.fc38.x86_64` is a stock kernel that does not contain the touchscreen drivers, so that would explain the last part.

I dont really know what could cause of the other issues you experienced. If it works on one app but not on another, that sounds like a GNOME issue tbh, same with the entire screen freezing and not even responding to the mouse.

**Comment by YosypC** [comment]
> `6.3.7-200.fc38.x86_64` is a stock kernel that does not contain the touchscreen drivers, so that would explain the last part. Our patched kernels have `.surface` as part of `uname -r`.

You are right! I chose the wrong kernel.

> I dont really know what could cause of the other issues you experienced. If it works on one app but not on another, that sounds like a GNOME issue tbh, same with the entire screen freezing and not even responding to the mouse.

I get many issues with the touch screen that I can't decide if the problem is one or another. Should I redirect my issues on GNOME's forums? I'm sure at least part of them is the kernel's fault, based on what I've read on journalctl.

**Comment by StollD** [comment]
> I get many issues with the touch screen that I can't decide if the problem is one or another. Should I redirect my issues on GNOME's forums? I'm sure at least part of them is the kernel's fault, based on what I've read on journalctl.

That is difficult to say. It really depends on the kind of issue. If its not related to gestures, you could always check if disabling iptsd fixes it, since that enables the fallback singletouch mode that is built into the hardware.

```
$ systemctl stop $(iptsd-find-service)
```

Or you could run iptsd-show after stopping iptsd so that you can see what iptsd "sees" and what it makes of the data that it receives from the hardware.

```
$ sudo iptsd-show $(iptsd-find-hidraw)
```

If you dont know if a problem is caused by GNOME or by the touchscreen drivers, better open a report here first.

**Comment by OriCat101** [comment]
I am having the aame issues. My SP 8 runs fine until it just freezes and becomes totally unresponsive. This does only happen if I use the touchscreen and not if I just use the cover keyboard and trackpad.

If I remember correctly, the issue started to occur in the last week.
If any more information is needed that I can provide, I'll happily provide it.

**Comment by TomReed123** [comment]
Similar results here on a SP8 and SP3. Mine do it when in tablet mode when attempting to size or move a window. System is still running in back ground. The SP8 works fine with the keyboard attached, however with it detached go to move a window and it locks up. What I can see is that the mouse pointer turns to a 4 headed arrow to move the window. When in tablet mode the arrow is not visible and there's no mouse pad to move the window. If I can provide further info please let me know.

**Comment by pernydev** [comment]
Same issue here with a SP7. Sometimes just freezes, even external mouse doesn't work.

**Comment by kdw2060** [comment]
I guess you're all suffering from the same Gnome/Mutter issue I and others reported here: https://gitlab.gnome.org/GNOME/mutter/-/issues/2897#note_1816064 
Since a few days another user came up with a workaround there.

@qzed , hope you don't mind mentioning you here. I can see you're the main contributor to this project and I was wondering if there is some contact between the maintainers of linux-surface and the developers of Gnome? Maybe linux-surface contributors could do QA on the touch-aspect of Gnome releases?

**Comment by qzed** [comment]
No worries, please do feel free to ping/@ me :)

I don't really have any direct contacts, apart from @jonas2515 who's done (doing?) some work for Gnome and has done some work here as well. I'm not really sure how QA like this would be set up, but we could have some thread/issue where Gnome people could ask for testing, or some list of volunteers to ping I guess. I'm open to ideas.

**Comment by StollD** [comment]
> Maybe linux-surface contributors could do QA on the touch-aspect of Gnome releases?

Maybe the GNOME developers should do that? :P (Only half kidding, KDE never seems to have these issues).

Our touchscreen drivers might be special snowflakes, but as far as GNOME is concerned there is no difference to a standard I2C or Wacom touchscreen. And frankly, there are enough devices that are supported upstream. 

Thats not to say you shouldn't use your surface to find and fix bugs on GNOME - hell no - if you want to do that, you absolutely should! But I feel like this isn't something that linux-surface the project should concern itself with.

**Comment by qzed** [comment]
> Our touchscreen drivers might be special snowflakes, but as far as GNOME is concerned there is no difference to a standard I2C or Wacom touchscreen. And frankly, there are enough devices that are supported upstream.

True. We do have a user-base with many touchscreen devices though, so I guess it might be easier to look for touchscreen users here if you want to test something with a larger group.

> Thats not to say you shouldn't use your surface to find and fix bugs on GNOME - hell no - if you want to do that, you absolutely should! But I feel like this isn't something that linux-surface the project should concern itself with.

Right. What I meant to say is: We can provide a platform for referring users interested in helping out to the Gnome team, but again, the organization/initiative would have to come from them (e.g. asking via an issue or Matrix/IRC, or something along this line). I do hope that they already do test their releases on touch devices regularly though.

**Comment by kdw2060** [comment]
You're absolutely right to expect this to be tested by the Gnome team, but judging by the reported issues I don't think they perform (rigorous) tests on touch devices. And nobody there seems to respond to the reported issues either (though luckily there's now a fellow user that has debugged the issue for them). 
So that's why I inquired if there's already some contact between the devs here and at Gnome.

I'd be happy to volunteer for testing new releases if this can be done in a VM (I'll post this on their Gitlab as well).
